src/handlers/rotate.rs

```diff
@@ -0,0 +1,126 @@
+use crate::{
+    util::{
+        auth::check_api_key,
+        image::{determine_img_dir, determine_img_path, save_image, ImageSearchBehaviour},
+    },
+    ServerState,
+};
+use axum::extract::{Query, State};
+use axum::headers::authorization::Bearer;
+use axum::headers::Authorization;
+use axum::http::StatusCode;
+use axum::TypedHeader;
+use libvips::{ops, VipsImage};
+use serde::Deserialize;
+use uuid::Uuid;
+
+#[derive(Deserialize)]
+pub struct RotateQuery {
+    id: Uuid,
+    angle: f64,
+}
+
+pub async fn rotate_handler(
+    State(server_state): State<ServerState>,
+    TypedHeader(authorization): TypedHeader<Authorization<Bearer>>,
+    query: Query<RotateQuery>,
+) -> Result<String, (StatusCode, String)> {
+    check_api_key(authorization, &server_state.api_key_hash)?;
+
+    if query.angle < 0.0 || query.angle > 360.0 {
+        return Err((
+            StatusCode::BAD_REQUEST,
+            "Angle must be 0.0 < angle < 360.0!".to_owned(),
+        ));
+    }
```

Comment:
What about `% 360`?

Reply:
It is better to notify the caller that the arguments are invalid in my opinion. I can't think of a case where a rotation outside of those bounds is valid / intended.

Reply:
Agreed.